pkgs/by-name/br/brother-dcpl2550dw/package.nix

```
@@ -0,0 +1,104 @@
+{ lib
+, stdenv
+, fetchurl
+, dpkg
+, autoPatchelfHook
+, makeWrapper
+, perl
+, gnused
+, ghostscript
+, file
+, coreutils
+, gnugrep
+, which
+}:
+
+let
+  arches = [ "x86_64" "i686" "armv7l" ];
+
+  runtimeDeps = [
+    ghostscript
+    file
+    gnused
+    gnugrep
+    coreutils
+    which
+  ];
+in
+
+stdenv.mkDerivation rec {
+  pname = "brother-dcpl2550dw";
+  version = "4.0.0-1";
+
+  nativeBuildInputs = [ dpkg makeWrapper autoPatchelfHook ];
+  buildInputs = [ perl ];
+
+  src = fetchurl {
+    url = "https://download.brother.com/welcome/dlf103577/dcpl2550dwpdrv-${version}.i386.deb";
+    sha256 = "01mrshcnr911bvd82xrb1sr6xybpsybzcdpjqs6896g56llbqrc0";
+  };
+
+  # unpackCmd = "dpkg-deb -x $src .";
+
+  patches = [
+    # The brother lpdwrapper uses a temporary file to convey the printer settings.
+    # The original settings file will be copied with "400" permissions and the "brprintconflsr3"
+    # binary cannot alter the temporary file later on. This fixes the permissions so the can be modified.
+    # Since this is all in briefly in the temporary directory of systemd-cups and not accessible by others,
+    # it shouldn't be a security concern.
+    ./fix-perm.patch
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cp -ar opt $out/opt
+    # delete unnecessary files for the current architecture
+  '' + lib.concatMapStrings
+    (arch: ''
+      echo Deleting files for ${arch}
+      rm -r "$out/opt/brother/Printers/DCPL2550DW/lpd/${arch}"
+    '')
+    (builtins.filter (arch: arch != stdenv.hostPlatform.linuxArch) arches) + ''
+    # bundled scripts don't understand the arch subdirectories for some reason
+    ln -s \
+      "$out/opt/brother/Printers/DCPL2550DW/lpd/${stdenv.hostPlatform.linuxArch}/"* \
+      "$out/opt/brother/Printers/DCPL2550DW/lpd/"
+
+    # Fix global references and replace auto discovery mechanism with hardcoded values
+    substituteInPlace $out/opt/brother/Printers/DCPL2550DW/lpd/lpdfilter \
+      --replace-quiet "my \$BR_PRT_PATH =" "my \$BR_PRT_PATH = \"$out/opt/brother/Printers/DCPL2550DW\"; #" \
+      --replace-quiet "PRINTER =~" "PRINTER = \"DCPL2550DW\"; #"
+    substituteInPlace $out/opt/brother/Printers/DCPL2550DW/cupswrapper/lpdwrapper \
+      --replace-quiet "my \$basedir = C" "my \$basedir = \"$out/opt/brother/Printers/DCPL2550DW\" ; #" \
+      --replace-quiet "PRINTER =~" "PRINTER = \"DCPL2550DW\"; #"
```

```suggestion
      --replace-fail "my \$BR_PRT_PATH =" "my \$BR_PRT_PATH = \"$out/opt/brother/Printers/DCPL2550DW\"; #" \
      --replace-fail "PRINTER =~" "PRINTER = \"DCPL2550DW\"; #"
    substituteInPlace $out/opt/brother/Printers/DCPL2550DW/cupswrapper/lpdwrapper \
      --replace-fail "my \$basedir = C" "my \$basedir = \"$out/opt/brother/Printers/DCPL2550DW\" ; #" \
      --replace-fail "PRINTER =~" "PRINTER = \"DCPL2550DW\"; #"
```
we want to notice if this no longer matches for whatever reason